### ISSUE TITLE
Swagger inheritance is broken

### ISSUE THREAD

**Issue by mnisius** [stoicflame/enunciate#1050]
It seems like the generated swagger file does not know about inheritance wich makes my generated client unusable. 

My example rest api looks like this:
We use immuables to generate pojos so please ignore that these are all interfaces
 
```java
//Main Input object for my rest service
public interface TestInput {
  LocalDateTime getCallTime();
  String getMessage();
  CommunicationLink getCommunicationLink();
}
public interface CommunicationLink {
  String commonField();
}
public interface Email extends CommunicationLink {
  String getEmailAddress();
}
public interface Telephone extends CommunicationLink {
  String getTelephonenumber();
}
```

I used the swagger.json to generate a java client. I tried to call my service like that: 

```java
TestInput input = new TestInput();
input.setCallTime(LocalDateTime.now());
input.setMessage("Hello World");
final Email email = new Email();
email.setCommonField("test");
email.setEmailAddress("test@test.com");
input.setCommunicationLink(email); //does not compile: Email does not extends CommunicationLink :-(
```

It seems like the inheritance is not translated into the generated swagger.json. 

This is what enunciate generated (only the relevant parts of the section "definitions") 
```json
  "CommunicationLink": {
    "type": "object",
    "title": "CommunicationLink",
    "example": {},
    "description": ""
  },
  "Email": {
    "type": "object",
    "title": "Email",
    "properties": {
      "commonField": {
        "readOnly": false,
        "description": "The value of the commonField attribute",
        "type": "string"
      },
      "emailAddress": {
        "readOnly": false,
        "description": "The value of the emailAddress attribute",
        "type": "string"
      }
    }
  },
  "Telephone": {
    "type": "object",
    "title": "Telephone",
    "properties": {
      "commonField": {
        "readOnly": false,
        "description": "The value of the commonField attribute",
        "type": "string"
      },
      "telephonenumber": {
        "readOnly": false,
        "description": "The value of the telephonenumber attribute",
        "type": "string"
      }
    }
  }
}
```

**Comment by mnisius** [comment]
I did another test. I replaced the interfaces with "real" objects. Now enunciate generates a diffrent swagger.json 
It has an allOf Section and references the supertype. But in the generated swagger client the inheritance is still not there so I get the same error as before. 

here is the generated swagger.json
```json
 "CommunicationLink": {
    "type": "object",
    "title": "CommunicationLink",
    "properties": {
      "commonField": {
        "readOnly": false,
        "description": "",
        "type": "string"
      }
    },
    "example": {
      "commonField": "..."
    },
    "description": ""
  },
  "Email": {
    "type": "object",
    "title": "Email",
    "allOf": [
      {
        "$ref": "#/definitions/CommunicationLink"
      },
      {
        "properties": {
          "emailAddress": {
            "readOnly": false,
            "description": "",
            "type": "string"
          }
        }
      }
    ],
    "example": {
      "emailAddress": "...",
      "commonField": "..."
    },
    "description": ""
  },
  "Telephone": {
    "type": "object",
    "title": "Telephone",
    "allOf": [
      {
        "$ref": "#/definitions/CommunicationLink"
      },
      {
        "properties": {
          "telephonenumber": {
            "readOnly": false,
            "description": "",
            "type": "string"
          }
        }
      }
    ],
    "example": {
      "telephonenumber": "...",
      "commonField": "..."
    },
    "description": ""
  }
```

**Comment by mnisius** [comment]
I think I found the solution. 

- Make Interfaces behave like classes (with allOf Sections) 
- Add this line to the super type:  ` "discriminator": "type",`

If I add this line manually the generated client uses inheritance as well and my code is working. 

I'm not sure why you tread interface inheritance diffrent than class inheritance. Is this because of this: https://github.com/stoicflame/enunciate/issues/374

So finally here is how the swagger.json definitions section should look like: 

```json
  "CommunicationLink": {
    "discriminator": "type",
    "type": "object",
    "title": "CommunicationLink",
    "properties": {
      "commonField": {
        "readOnly": false,
        "description": "",
        "type": "string"
      }
    },

  },
  "Email": {
    "type": "object",
    "title": "Email2",
    "allOf": [
      {
        "$ref": "#/definitions/CommunicationLink"
      },
      {
        "properties": {
          "emailAddress": {
            "readOnly": false,
            "description": "",
            "type": "string"
          }
        }
      }
    ]
  },
  "Telephone": {
    "type": "object",
    "title": "Telephone",
    "allOf": [
      {
        "$ref": "#/definitions/CommunicationLink"
      },
      {
        "properties": {
          "telephonenumber": {
            "readOnly": false,
            "description": "",
            "type": "string"
          }
        }
      }
    ]
  }
```

**Comment by stoicflame** [comment]
Fixed at 5184d70f0.

Note that the interface "superclass" (in this case `CommunicationLink` will need to be annotated with `@JsonTypeInfo` in order for the discriminator to be determined and applied.

**Comment by stoicflame** [comment]
Fixed in `2.13.2`.
